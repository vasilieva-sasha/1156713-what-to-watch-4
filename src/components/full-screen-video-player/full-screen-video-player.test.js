import React from "react";
import renderer from 'react-test-renderer';
import FullScreenVideoPlayer from "./full-screen-video-player";
import configureStore from 'redux-mock-store';
import NameSpace from './../../reducer/name-space';
import {Provider} from "react-redux";

const mock = {
  id: 1,
  title: ``
};

describe(`FullScreenVideoPlayerComponent`, () => {
  const mockStore = configureStore([]);
  const store = mockStore({
    [NameSpace.APP]: {
      genre: `All genres`,
    },
    [NameSpace.USER]: {
      authorizationStatus: `NO_AUTH`,
    },
  });
  it(`Full Screen Video Player renders correctly while playing`, () => {
    const tree = renderer.create(
        <Provider store={store}>
          <FullScreenVideoPlayer onExitClick={() => {}}
            isPlaying={true}
            renderVideo={() => {}}
            renderPlayButton={() => {}}
            renderPauseButton={() => {}}
            getTogglerPosition={() => {}}
            getCountDown={() => {}}
            onFullScreenClick={() => {}}
            film={mock}
          />
        </Provider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Full Screen Video Player renders correctly while pause`, () => {
    const tree = renderer.create(
        <Provider store={store}>
          <FullScreenVideoPlayer onExitClick={() => {}}
            isPlaying={false}
            renderVideo={() => {}}
            renderPlayButton={() => {}}
            renderPauseButton={() => {}}
            getTogglerPosition={() => {}}
            getCountDown={() => {}}
            onFullScreenClick={() => {}}
            film={mock}
          />
        </Provider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
