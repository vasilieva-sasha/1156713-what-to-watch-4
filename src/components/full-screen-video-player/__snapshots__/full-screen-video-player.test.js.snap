// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FullScreenVideoPlayerComponent Full Screen Video Player renders correctly while pause 1`] = `
<div
  className="player"
>
  <button
    className="player__exit"
    onClick={[Function]}
    type="button"
  >
    Exit
  </button>
  <div
    className="player__controls"
  >
    <div
      className="player__controls-row"
    >
      <div
        className="player__time"
      >
        <progress
          className="player__progress"
          max="100"
        />
        <div
          className="player__toggler"
          style={
            Object {
              "left": "undefined%",
            }
          }
        >
          Toggler
        </div>
      </div>
      <div
        className="player__time-value"
      />
    </div>
    <div
      className="player__controls-row"
    >
      <div
        className="player__name"
      >
        Transpotting
      </div>
      <button
        className="player__full-screen"
        onClick={[Function]}
        type="button"
      >
        <svg
          height="27"
          viewBox="0 0 27 27"
          width="27"
        >
          <use
            xlinkHref="#full-screen"
          />
        </svg>
        <span>
          Full screen
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`FullScreenVideoPlayerComponent Full Screen Video Player renders correctly while playing 1`] = `
<div
  className="player"
>
  <button
    className="player__exit"
    onClick={[Function]}
    type="button"
  >
    Exit
  </button>
  <div
    className="player__controls"
  >
    <div
      className="player__controls-row"
    >
      <div
        className="player__time"
      >
        <progress
          className="player__progress"
          max="100"
        />
        <div
          className="player__toggler"
          style={
            Object {
              "left": "undefined%",
            }
          }
        >
          Toggler
        </div>
      </div>
      <div
        className="player__time-value"
      />
    </div>
    <div
      className="player__controls-row"
    >
      <div
        className="player__name"
      >
        Transpotting
      </div>
      <button
        className="player__full-screen"
        onClick={[Function]}
        type="button"
      >
        <svg
          height="27"
          viewBox="0 0 27 27"
          width="27"
        >
          <use
            xlinkHref="#full-screen"
          />
        </svg>
        <span>
          Full screen
        </span>
      </button>
    </div>
  </div>
</div>
`;
